### DISCUSSION TITLE
Consider renaming to "spenserblack/publish-wiki-action"

### DISCUSSION THREAD

**Discussion by jcbhmr** [spenserblack/actions-wiki#17] [Ideas]
I think this is a relatively cosmetic-level change. There's 301 redirects, so nothing would change for folks still using spenserblack/actions-wiki

This is largely motivated by branding, and SEO and all those good advertising things. And just good naming conventions in general. To quote myself

> A lot of the names are honestly kind of vague. something like "actions/publish-wiki" is short and punchy, but since we aren't in the https://github.com/actions org, the next best thing is name/actions-publish-wiki maybe? Or publish-wiki-action? I'm basing this naming convention on the existing actions/* projects like "actions/deploy-pages" and "actions/configure-pages" and "actions/checkout". See https://github.com/jcbhmr/publish-wiki-action/issues/12 for more.
> 
> My personal favorite name would probably be "name/publish-wiki-action".
> 
> — https://github.com/jcbhmr/publish-wiki-action/issues/8#issuecomment-1461120686

> # publish-wiki-action
> or if possible
> 
> # some-actions-org/publish-wiki

https://github.com/jcbhmr/publish-wiki-action-OLD/issues/12

This is **an opinionated suggestion** that you can close and I'll accept that. This is just a _suggestion_ of my own thoughts on the naming stuff.

**Comment by spenserblack** [comment]
:thinking: I think I might prefer to keep the name short. I understand that `actions-wiki` is a bit vague and doesn't have a verb like "publish", but given how wordy the action is with my full name already in the prefix, I'd like to reduce those few characters to the absolute minimum.

My other actions all have the `actions-` prefix, so at the very least I would keep that.

**Comment by jcbhmr** [comment]
>  doesn't have a verb like "publish"

Yeah this is the biggest thing in my book. Verbs are awesome to get a quick idea of what something does. I remember that it was a literal _rule_ in some of my compsci classes that all functions have verbs in their names. It's actually a good convention. https://stackoverflow.com/a/1991341

> given how wordy the action is with my full name already in the prefix, I'd like to reduce those few characters to the absolute minimum.

I can understand this. That's why I renamed my action first from jcbhmr/publish-to-github-wiki to jcbhmr/publish-wiki but that didn't have "action" in it, so I added the suffix to seem like a vaguely english title. I guess. That worked for me. I accept that you have different preferences.

> My other actions all have the actions- prefix, so at the very least I would keep that.

Understandable. I accept that, but I don't think history should _necessitate_ or _require_ that such a trend continue ad infinitum. break the mold lmao.

---

I guess if you find my argument persuasive enough, there's never a better time than now to make change. But I understand that as-is is way easier and "safer". So yeah. I'll close this but I guess at least I proposed the idea.

Thanks for explaining your rationale! 👍 ❤ _btw no hard feelings 🤣_

**Comment by jcbhmr** [comment]
As further evidence for the -action suffix vs actions- prefix, I present statistical popularity of the naming conventions:

https://sourcegraph.com/search?q=context:global+repo:actions-+count:all&patternType=standard&sm=1&groupBy=repo
![image](https://user-images.githubusercontent.com/61068799/225121215-44fe638e-ef8d-4487-bd7c-45c320f8fbf2.png)

https://sourcegraph.com/search?q=context:global+repo:-action+count:all&patternType=standard&sm=0&groupBy=repo
![image](https://user-images.githubusercontent.com/61068799/225121174-9e1004d1-cec1-4c03-a380-e1879d714622.png)

I'm sorry I'm so stuck on this but it's festering in my ocd brain. 🤣

**Comment by jcbhmr** [comment]
Also this would **greatly** help Google SEO indexing because right now it's using the  _literal repo slug_ as the title. Hence, a more descriptive, **even if it's longer** title would be a good idea.

>  given how wordy the action is with my full name already in the prefix, I'd like to reduce those few characters to the absolute minimum.

I don't think this should be a problem. Look at how successful Andrew-Chen-Wang/github-wiki-action is! It's like 10 more chars vs the current name. Also how many times are you typing this name? **Once.** That's it. In one workflow file, and then _never again_ probably. It's not like a `std::` prefix that needs to be efficient instead of `standard_library::` 👨‍💻 Point is that I don't think this should be a reason/excuse. _Laziness_ or _conformity with my other actions_, now _that's_ a reason! (or an excuse) 😎

![image](https://user-images.githubusercontent.com/61068799/225122410-6f42c690-a25e-4927-9cac-e1e2d1f8b613.png)

I think this SEO thing is a compelling enough reason to pursue a rename. Now's the perfect time to do it while it's 🆕

**opinion** Also if I didn't comment on it before: the -action suffix is more "English-y".

**Comment by jcbhmr** [comment]
I guess I can live with the `actions-` prefix. 🤷‍♂️ I would like to see something like `actions-publish-wiki` though to improve the name recognition and help SEO.

**Reply by jcbhmr** [reply]
@spenserblack  Is there any chance of this compromise actually happening, or should I just give up on this topic and never mention it again? 😆

**Reply by spenserblack** [reply]
Yeah, probably not going to rename. Or, at the very least, I'll hesitate/stall for a while :laughing:

**Comment by jcbhmr** [comment]
waitaminute what if you did
1. rename actions-wiki to publish-wiki-action
2. rename publish-wiki-action back to actions-wiki

this would get the gh alias redirect thing where I could use spenserblack/publish-wiki-action or whatever and it would work!

just an idea tho 🤷‍♀️

**Reply by spenserblack** [reply]
Interesting idea, maybe I'll try it out 🙂
